plugins/catalog-backend/src/search/defaultCatalogCollatorEntityTransformer.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, isGroupEntity, isUserEntity } from '@backstage/catalog-model';
+import {
+  CatalogCollatorEntityTransformer,
+  TransformedEntity,
+} from './CatalogCollatorEntityTransformer';
+
+const getDocumentText = (entity: Entity): string => {
+  const documentTexts: string[] = [];
+  documentTexts.push(entity.metadata.description || '');
+
+  if (isUserEntity(entity) || isGroupEntity(entity)) {
+    if (entity.spec?.profile?.displayName) {
+      documentTexts.push(entity.spec.profile.displayName);
+    }
+  }
+
+  return documentTexts.join(' : ');
+};
+
+export const defaultCatalogCollatorEntityTransformer: CatalogCollatorEntityTransformer =
+  (entity: Entity): TransformedEntity => {
```

Comment:
you can make this just `entity => {` since the types are redundant; it's all embedded already in `CatalogCollatorEntityTransformer` on the previous row

Reply:
actually adding return type explicitly makes types more strict, e.g. if you try this out you will find that without explicit return type you can add any extra field and types will no say you anything while with the return type TS will shout on you

So basically I want to suggest to add return type explicitly everywhere and add lint rule for that

Reply:
I don't mind though. I'd say that it's fine to have extra fields as long as you fulfil the contract.

Reply:
I think in this particular thing it will lead to adding some extra data to the document without any complaint and here it's not that good as it will lead to allocating extra space in memory etc. So from my perspective some additional strictness here should be

Reply:
We don't do that anywhere else though. And whether agreeing with that or not, I'd say that getting rid of the type in the contract is the bigger deal according to me.

Reply:
done